src/main/java/com/activeviam/tools/bookmark/impl/JsonUiToContentServer.java

```diff
@@ -109,44 +92,49 @@ private static File getSubDirectory(final String folderName) {
    *
    * @param root the current directory to add to the structureTree.
    */
-  private static SnapshotContentTree createDirectoryTree(File root, SnapshotContentTree parent) {
-    if (root == null) {
-      LOGGER.info("Nothing to load");
-      return null;
-    }
-    for (File child : root.listFiles()) {
-      if (child.isFile()) {
-        final JsonNode jsonNodeContent = loadFileIntoNode(child);
-        final SnapshotContentTree node =
-            new SnapshotContentTree(
-                jsonNodeContent.toString(),
-                false,
-                PERMISSIONS.get(ContentServerConstants.Role.OWNERS),
-                PERMISSIONS.get(ContentServerConstants.Role.READERS),
-                new HashMap<>());
-        parent.putChild(child.getName().replace(Paths.JSON, ""), node, true);
-      } else {
-        final SnapshotContentTree childNode = createEmptyDirectoryNode();
-        SnapshotContentTree childTree = createDirectoryTree(child, childNode);
-        parent.putChild(child.getName(), childTree, true);
+  private static SnapshotContentTree createDirectoryTree(String root, SnapshotContentTree parent) {
+    try {
+      Resource[] rootFile = dashboardTreeResolver.getResources("classpath*:/**/" + root + "/*");
+      for (Resource child : rootFile) {
+        String[] path = child.getURL().toString().split(Paths.SEPARATOR);
+        String childName = path[path.length - 1];
+        if (childName.endsWith(Paths.JSON)) {
+          final JsonNode jsonNodeContent = loadFileIntoNode(child.getInputStream());
+          final SnapshotContentTree node =
+              new SnapshotContentTree(
+                  jsonNodeContent.toString(),
+                  false,
+                  PERMISSIONS.get(ContentServerConstants.Role.OWNERS),
+                  PERMISSIONS.get(ContentServerConstants.Role.READERS),
+                  new HashMap<>());
+          parent.putChild(childName.replace(Paths.JSON, ""), node, true);
+        } else {
+          final SnapshotContentTree childNode = createEmptyDirectoryNode();
+          SnapshotContentTree childTree =
+              createDirectoryTree(root + Paths.SEPARATOR + childName, childNode);
+          parent.putChild(childName, childTree, true);
+        }
       }
+      return parent;
+    } catch (Exception ioe) {
```

Comment:
Could we have a more precise exception type?

Reply:
yes, it is supposed to be an IOException